Modules/main.c

```diff
@@ -597,6 +597,9 @@ pymain_run_python(int *exitcode)
 
     pymain_header(config);
 
+    _PyInterpreterState_SetRunningMain(interp);
```

Comment:
Is this really the only place this should be called? What about users embedding Python and executing 'main' their own way?

Reply:
Yeah, they could call this too.